connector/src/test/scala/com/basho/riak/spark/rdd/ReadFromRiakRDDTest.scala

```diff
@@ -292,4 +301,31 @@ class ReadFromRiakRDDTest extends AbstractRDDTest{
 
     assertEquals(7, allValues.size)
   }
+
+  @Test
+  def readDataUsingCustomConverterFactory(): Unit = {
+    // The aim of this custom converter is to extract string timestamp values from UserTS objects and convert them
+    // to long values. In other words it converts UserTS.timestamp -> UserTsTimestamp
+    ConverterFactory.getInstance.registerConverterForClass(classOf[UserTSTimestamp], new Converter[UserTSTimestamp](classOf[UserTSTimestamp]) {
+      override def toDomain(binaryValue: BinaryValue, s: String): UserTSTimestamp = {
+        val jsonConverter = ConverterFactory.getInstance.getConverter(classOf[UserTS])
```

Comment:
Better to add comment kinda: "Using default Json conversion from java clietn to.. bla bla bla"